src/mlpack/methods/adaboost/adaboost_impl.hpp

```diff
@@ -211,26 +211,59 @@ void AdaBoost<WeakLearnerType, MatType>::Classify(
     arma::Row<size_t>& predictedLabels)
 {
   arma::Row<size_t> tempPredictedLabels(test.n_cols);
-  arma::mat cMatrix(numClasses, test.n_cols);
+  arma::mat probabilities;
+  
+  probabilities.zeros(numClasses, test.n_cols);
+  predictedLabels.set_size(test.n_cols);
+
+  for (size_t i = 0; i < wl.size(); i++)
+  {
+    wl[i].Classify(test, tempPredictedLabels);
+
+    for (size_t j = 0; j < tempPredictedLabels.n_cols; j++)
+      probabilities(tempPredictedLabels(j), j) += alpha[i];
+  }
+
+  arma::colvec pRow;
+  arma::uword maxIndex = 0;
+
+  for (size_t i = 0; i < predictedLabels.n_cols; i++)
+  {
+    pRow = probabilities.unsafe_col(i);
+    pRow.max(maxIndex);
+    predictedLabels(i) = maxIndex;
+  }
```

Comment:
I think you could simplify this to just use the other overload; here's an example implementation:

```
arma::mat probabilities;
Classify(test, predictedLabels, probabilities);
```

Let me know if I overlooked something. :+1:

Reply:
There isn't such overload which can take probabilities as an argument.

Reply:
> There isn't such overload which can take probabilities as an argument.

Yes , there isn't one, instead you have to implement it :)

Reply:
My bad, I actually did implemented it and thought he was talking about some other overload.